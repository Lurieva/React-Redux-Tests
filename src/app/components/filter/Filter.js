import React  from 'react';
import PropTypes from 'prop-types';

import { Input, Button } from '../../shared/';
import SearchBy from './searchBy/SearchBy';

import './filter.scss';

const Filter = ({ searchBy, onApplyFilter, onChangeSearchBy, onFilterChange }) => (
    <div className="filter">
        <div className="filter__title">Find your movie</div>
        <Input className="filter__input" onChange={onFilterChange} enterClick={onApplyFilter}/>
        <div className="filter__action">
            <SearchBy searchBy={searchBy} onChange={onChangeSearchBy}/>
            <Button name="Search" className="filter__action__apply-btn" onClick={onApplyFilter}/>
        </div>
    </div> 
)

Filter.propTypes = {
    searchBy: PropTypes.string.isRequired,
    onFilterChange: PropTypes.func.isRequired,
    onChangeSearchBy: PropTypes.func.isRequired,
    onApplyFilter: PropTypes.func.isRequired
};

export default Filter;