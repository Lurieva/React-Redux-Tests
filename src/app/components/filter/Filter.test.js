import React from 'react';
import { shallow } from 'enzyme';
import Filter from './Filter';

jest.mock('../../shared/Input/Input', () => 'Input');
jest.mock('../../shared/Button/Button', () => 'Button');
jest.mock('./searchBy/SearchBy', () => 'SearchBy');

describe('Filter', () => {
    const mockFn = jest.fn();
    const searchBy = 'value';

    it('should be defined', () => {
        expect(Filter).toBeDefined();
    });

    it('should render correctly ', () => {
        const tree = shallow(<Filter searchBy={searchBy}
                                     onClick={mockFn}
                                     onChangeSearchBy={mockFn}
                                     onFilterChange={mockFn} />);
        expect(tree).toMatchSnapshot();
    });
});