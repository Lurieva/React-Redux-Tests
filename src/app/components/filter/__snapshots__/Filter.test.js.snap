// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter
    onChangeSearchBy={[MockFunction]}
    onClick={[MockFunction]}
    onFilterChange={[MockFunction]}
    searchBy="value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="filter__title"
        >
          Find your movie
        </div>,
        <Input
          className="filter__input"
          enterClick={undefined}
          onChange={[MockFunction]}
        />,
        <div
          className="filter__action"
        >
          <SearchBy
            onChange={[MockFunction]}
            searchBy="value"
          />
          <Button
            className="filter__action__apply-btn"
            name="Search"
            onClick={undefined}
          />
        </div>,
      ],
      "className": "filter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find your movie",
          "className": "filter__title",
        },
        "ref": null,
        "rendered": "Find your movie",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "filter__input",
          "enterClick": undefined,
          "onChange": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": "Input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SearchBy
              onChange={[MockFunction]}
              searchBy="value"
            />,
            <Button
              className="filter__action__apply-btn"
              name="Search"
              onClick={undefined}
            />,
          ],
          "className": "filter__action",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [MockFunction],
              "searchBy": "value",
            },
            "ref": null,
            "rendered": null,
            "type": "SearchBy",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "filter__action__apply-btn",
              "name": "Search",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "Button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="filter__title"
          >
            Find your movie
          </div>,
          <Input
            className="filter__input"
            enterClick={undefined}
            onChange={[MockFunction]}
          />,
          <div
            className="filter__action"
          >
            <SearchBy
              onChange={[MockFunction]}
              searchBy="value"
            />
            <Button
              className="filter__action__apply-btn"
              name="Search"
              onClick={undefined}
            />
          </div>,
        ],
        "className": "filter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find your movie",
            "className": "filter__title",
          },
          "ref": null,
          "rendered": "Find your movie",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "filter__input",
            "enterClick": undefined,
            "onChange": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": "Input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SearchBy
                onChange={[MockFunction]}
                searchBy="value"
              />,
              <Button
                className="filter__action__apply-btn"
                name="Search"
                onClick={undefined}
              />,
            ],
            "className": "filter__action",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [MockFunction],
                "searchBy": "value",
              },
              "ref": null,
              "rendered": null,
              "type": "SearchBy",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "filter__action__apply-btn",
                "name": "Search",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "Button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
